src/Compilers/CSharp/Portable/Binder/ContextualAttributeBinder.cs

```diff
@@ -64,5 +66,10 @@ internal Symbol AttributeTarget
                 return _attributeTarget;
             }
         }
+
+        protected override CSDiagnosticInfo SingleTypeNameNotFound(string whereText, DiagnosticBag diagnostics, Location location, AssemblySymbol forwardedToAssembly)
+        {
+            return base.SingleTypeNameNotFound($"{whereText}' or '{whereText}Attribute", diagnostics, location, forwardedToAssembly);
```

Comment:
It seems the compiler in VS2008 used the or approach. Testing VS2010 now, I see multiple errors emitted, which solves the localizable issue. Emitting multiple errors seems a bit trickier to do from within NotFound, but I will have a look.